citus.spec

```diff
@@ -34,8 +34,24 @@ commands.
 %prep
 %setup -q -n %{sname}-%{version}
 
+# Flags taken from: https://liquid.microsoft.com/Web/Object/Read/ms.security/Requirements/Microsoft.Security.SystemsADM.10203#guide
+SECURITY_CFLAGS="-fstack-protector-strong -D_FORTIFY_SOURCE=2 -O2 -z noexecstack -fpic -Wl,-z,relro -Wl,-z,now -Wformat -Wformat-security -Werror=format-security"
+
+currentgccver="$(gcc -dumpversion)"
+requiredgccver="4.8.2"
+if [ "$(printf '%s\n' "$requiredgccver" "$currentgccver" | sort -V | tail -n1)" = "$requiredgccver" ]; then
+      echo WARNING: Using slower security flags because of outdated compiler
+      SECURITY_CFLAGS="-fstack-protector-all -D_FORTIFY_SOURCE=2 -O2 -z noexecstack -fpic -Wl,-z,relro -Wl,-z,now -Wformat -Wformat-security -Werror=format-security"
+    fi
+fi
+
+gccgte8=$(expr `gcc -dumpversion | cut -f1 -d.` \>= 8)
```

Comment:
I'd prefer if this used the same type of check as the one above for consistency, i.e. using `sort -V` (version sort).

Reply:
Actually thinking about that a bit, we would enter that if it the version is equal to "4.8.2" but we don't want to, isn't that if check wrong in this case?

Reply:
I guess you are right, maybe a better way would be to use `head -n1` and check against currentgccver

Reply:
if currentgccver and requiredgccver are the same, then what is the difference between head -n1 or tail -n1, or checking against `currentgccver` or `requiredgccver`. I would say that we can switch to `gte` structure here

Reply:
```bash
if ! [ "$(printf '%s\n' "$requiredgccver" "$currentgccver" | sort -V | head -n1)" = "$requiredgccver" ]; then
      echo WARNING: Using slower security flags because of outdated compiler
fi
```

or something like the above maybe

Reply:
yes that seems good, but maybe use `!=`  instead of `! [ ... = ... ]`

And then use 
```
if [ "$(printf '%s\n' "8.0.0" "$currentgccver" | sort -V | head -n1)" = "8.0.0" ]; then
      # add extra security flag
fi